src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -5,10 +5,113 @@
 
 'use strict'
 
+import * as child_process from 'child_process'
 import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
 import { fileExists } from '../../filesystemUtilities'
-import { DefaultSamCliTaskInvoker } from './samCliInvoker'
-import { SamCliTaskInvoker } from './samCliInvokerUtils'
+import { ChildProcess } from '../../utilities/childProcess'
+import { ChannelLogger } from '../../utilities/vsCodeUtils'
+
+const localize = nls.loadMessageBundle()
+
+export const WAIT_FOR_DEBUGGER_MESSAGES = {
+    PYTHON: 'Waiting for debugger to attach...',
```

Comment:
IIRC I now see 'waiting for debugger to attach...' from Node and .NET as well.

Reply:
There is no guarantee that each runtime uses the same message in SAM.

We happen to control the Python message since we generate the debug wrapper, but node and dotnet appear to be different.

Reply:
We have to be aware of current and former messages also, up to our min supported version.

Reply:
If the messages are similar, would it make more sense to have:

```typescript
export const WAIT_FOR_DEBUGGER_MESSAGES = {
    PYTHON: 'Waiting for debugger to attach...',
    NODEJS810: 'waiting for debugger to attach',
}
```

Reply:
Discussion has determined that the containers used by SAM are the ones from https://github.com/lambci/docker-lambda/ . We'll follow up there in a separate thread; this is good for now.

Reply:
what are these messages actually used for?

Reply:
This PR launches `sam local invoke` as a child process. When debugging locally, we watch the child process' stderr for these messages. We do not attempt to attach a debugger to the sam app until we've seen one of these messages.

Reply:
@kiiadi We use the messages to determine when the container is ready for the debugger attached. Previously we used a combination of port scanning + retry loops. This was unstable, inconsistent between languages, relied on assumptions about undocumented VS Code behavior, and was broken by by an update to VS Code.

Output scanning isn't elegant, but it has been the only robust solution in our testing. It's also similar to VS Code's built-in mechanism for attaching to a background task. See [here](https://github.com/aws/aws-toolkit-vscode/blob/develop/docs/debugging-nodejs-lambda-functions.md#optional-automatically-start-debugging-when-ready) for an example.

Reply:
side-bar convo with @awschristou - I get it now! thanks.